File: app/src/main/java/com/katkov/lolachievements/application/ui/firstentry/FirstEntryView.java
package com.katkov.lolachievements.application.ui.firstentry;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

public interface FirstEntryView extends MvpView {

    @StateStrategyType(OneExecutionStateStrategy.class)
    void showServerChoiceDialog();

    @StateStrategyType(AddToEndSingleStrategy.class)
    void showSelectedName(String selectedName);
}

File: app/src/main/java/com/katkov/lolachievements/application/ui/firstentry/FirstEntryView.kt
package com.katkov.lolachievements.application.ui.firstentry

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

interface FirstEntryView : MvpView {

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun showServerChoiceDialog()

    @StateStrategyType(AddToEndSingleStrategy::class)
    fun showSelectedName(selectedName: String)
}
